### PR TITLE
Rhmap 2458 internet datasources

### PR DESCRIPTION
**Pull request by lholmquist**
This adds endpoints for accessing data from a source on the internet.  

There are currently 2 endpoints.
- A List of Hobbies - http://www.notsoboringlife.com/list-of-hobbies/
- List of Star Wars films - https://swapi.co/

### CODE REVIEW

**Comment by lholmquist** [comment]
@nialldonnellyfh

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (nialldonnellyfh) [inline .jshintrc]: Would be good to populate the jshint (e.g. https://github.com/fheng/fh-supercore/blob/master/.jshintrc)

File: README.md [hunk 2]
```diff
@@ -1 +1 @@
-# appforms-datasources-internet
+Feedhenry Sample Service Using data from the Internet for Datasources
```
Comment (nialldonnellyfh) [inline README.md]: Can you expand on this readme to include a summary of what is happening in the service.

File: lib/hobbies.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
+var express = require('express');
+var bodyParser = require('body-parser');
+var cors = require('cors');
+var request = require('request');
+var $ = require('cheerio');
+
+
+function hobbiesRoute() {
```
Comment (nialldonnellyfh) [inline lib/hobbies.js]: I don't think you need the function here. You can just create a router an export the router directly.

File: lib/hobbies.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  var hobbiesRouter = new express.Router();
```
Comment (nialldonnellyfh) [inline lib/hobbies.js]: Don't need the "new" here. Express will do it for you.

(http://expressjs.com/4x/api.html#router)

File: lib/hobbies.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+      // On the page, the list of hobbies is an ordered list <ol>, with a class of 'list-of-hobbies'
+      // ex:  <ol class="list-of-hobbies">...</ol>
+      // So we will search for this list by the classname
+
+      // And then we wil parse it into the DataSource format
+
+      var parsedHobbies = parsedHTML('ol.list-of-hobbies li.loh_column1');
+
+      var listOfHobbies = [];
```
Comment (nialldonnellyfh) [inline lib/hobbies.js]: You might be interested in the underscore library here. I find it really useful for dealing with iterative stuff.

http://underscorejs.org/#map

File: lib/hobbies.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+      parsedHobbies.map(function  (index, el) {
+        listOfHobbies.push({
+          key: index,
+          value: $(el).text()
+        });
+      });
+
+      // And then return it
+      res.send(listOfHobbies);
```
Comment (nialldonnellyfh) [inline lib/hobbies.js]: res.json is probably better here if you want to return a JSON response.

(http://expressjs.com/4x/api.html#res.json)

File: lib/sw-movies.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
+var express = require('express');
+var bodyParser = require('body-parser');
+var cors = require('cors');
+var swapi = require('swapi-node');
+
+
+function swRoute() {
```
Comment (nialldonnellyfh) [inline lib/sw-movies.js]: Same as above. I don't think you need the function here. You can just export the router directly. Makes for cleaner code.

File: lib/sw-movies.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+  var swRouter = new express.Router();
```
Comment (nialldonnellyfh) [inline lib/sw-movies.js]: Don't need the "new" here.

File: lib/sw-movies.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+  swRouter.use(cors());
+  swRouter.use(bodyParser());
+
+  function handleError(err, response){
```
Comment (nialldonnellyfh) [inline lib/sw-movies.js]: Might want to abstract this error handler out. Repetition here.

File: lib/sw-movies.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+        };
+      });
+
+      // Now lets do a simple sort here to get the movies in order.
+      listOfMovies = listOfMovies.sort(function (a, b) {
+        return a.key > b.key;
+      });
+
+      // Now we return the list
+      res.send(listOfMovies);
```
Comment (nialldonnellyfh) [inline lib/sw-movies.js]: res.json would be good here.

**Comment by nialldonnellyfh** [comment]
Reviewed. Some comments above. Also can you rebase and squash your commits to a single one.

File: lib/hobbies.js (hunk 5 above)
Comment (lholmquist) [inline lib/hobbies.js]: yea,  i usually lodash or the native Array.prototype.map,  but this map is actually part of the cheerio lib and acts slightly differently.  i've updated the code though with a more streamlined usage of it

**Comment by lholmquist** [comment]
@nialldonnellyfh made a few of the updates.  If this looks good,  i will merge

File: README.md [hunk 11]
```diff
@@ -1 +1,32 @@
-# appforms-datasources-internet
+# Feedhenry Sample Service Using data from the Internet for Datasources
```
Comment (nialldonnellyfh) [inline README.md]: "for Datasources" should be "for Data Sources"

File: README.md [hunk 12]
```diff
@@ -1 +1,32 @@
...
+
+This sample RESTful service has 2 endpoints, `/hobbies` and `/sw` for loading data from the Internet and then properly formatting it as a Data Source
+
+The `/hobbies` endpoint will "scrap" the website,  http://www.notsoboringlife.com/list-of-hobbies/, parse the HTML, and produce a list.
```
Comment (nialldonnellyfh) [inline README.md]: will "scrap" the website -> scrape?

File: package.json [hunk 13]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "fh-datasources-internet-cloud",
+  "version": "0.0.1",
+  "description": "Feedhenry Sample Service Using Data from the Internet for Datasources",
```
Comment (nialldonnellyfh) [inline package.json]: Internet for Datasources ->  Internet for Data Sources

**Comment by nialldonnellyfh** [comment]
Couple of small comments. Otherwise looks good to me. Feel free to merge back when finished.

File: README.md (hunk 12 above)
Comment (lholmquist) [inline README.md]: dude, english is hard  :)

**Pull request merged**